### ISSUE TITLE
addThemesPane() is not a function

### ISSUE THREAD

**Issue by PartialDevAdmin** [PartialDevAdmin/theme-changer#2]
When browsing to /admin in my forum, I get the error

`addThemesPane()` is not a function`

I am assuming this error is coming from the `main.js` file, which looks like

```
import { extend } from 'flarum/extend';
import Page from 'reflar/theme-changer/models/Page';
import addThemesPane from 'reflar/theme-changer/addThemesPane';

app.initializers.add('reflar-theme-changer', app => {
  app.store.models.pages = Page;
  addThemesPane();
});
```

Any ideas?

**Comment by datitisev** [comment]
Does the addThemsePane.js exist, and is it located in the same folder as main.js?

**Comment by datitisev** [comment]
Then you probably messed up the gulp file. Transfer this repository to the ReFlar organization, please.

**Comment by PartialDevAdmin** [comment]
Ok

**Comment by PartialDevAdmin** [comment]
See any issues @datitisev?

**Comment by CDK2020** [comment]
I see a few potential issues try them in this order:

1. It looks like on line 7 of main.js you don't call it as a function.
2. On line 5 of main.js you have the initializer as "reflar.theme-changer" change it to "reflar-theme-changer" (. to -) 
3. Use the full path (reflar/theme-changer/addThemesPane) instead of the shorthand (./addThemesPane)

**Comment by Ralkage** [comment]
@PartialDevAdmin in short, you are calling `addThemesPane` as a variable instead of an actual function such as `addThemesPane();`.

**Comment by PartialDevAdmin** [comment]
Yay! That fixed it! I click on it, and nothing happens, but that is just minor for right now.

**Comment by PartialDevAdmin** [comment]
Alright! The base code for the admin page is in. @CDK2020 @datitisev Can you give me some tips for pulling data and importing it into the page?

**Comment by Ralkage** [comment]
@CDK2020 would be your relationships expert as this part also focuses on models and such 👍

**Comment by PartialDevAdmin** [comment]
@Ralkage Ok, thanks! I will wait for a reply from him. I can just taste it ;)

**Comment by CDK2020** [comment]
Thankfully we won't have to use then here friggin hate relationships). The best way to do it is make an API endpoint for the data (unless the data is on the settings table then do something different).

**Comment by PartialDevAdmin** [comment]
Ok. It looks like the pages extension that I borrowed code from uses an API to pull information, so I am going to take a look there. Any suggestions or tips @CDK2020?

**Comment by PartialDevAdmin** [comment]
Also, I changed the icon to the FA icon `paint-brush` and now it looks just like the icon on the Appearance page. Should I change it to something else?

**Comment by CDK2020** [comment]
You should for sure change the icon, maybe add "-o" to the end of the icon (may or may not work) I have no idea what data you are trying to pull and from where so I'd have to see that first. Also, POST is for putting data initially, PATCH I'd for changing data, then there is GET and DELETE

**Comment by PartialDevAdmin** [comment]
Ok, Thanks! I will see what I can drum up... drum roll lol

**Comment by PartialDevAdmin** [comment]
@CDK2020 Just pushed some major updates, can you see any errors in the code? I have not enabled the API yet so I do not know if it works, just wondering if you can spot any errors off-hand

**Comment by CDK2020** [comment]
I have no idea, try to run it and I'll take a look if you find errors

**Comment by PartialDevAdmin** [comment]
Ok, will do!

**Comment by CDK2020** [comment]
One thing I do see is that you don't have a theme modal class, nor do you have migrations to make a theme table

**Comment by PartialDevAdmin** [comment]
Yup, migrations are coming right away, just changing the api and other command thingies namespaces over and renaming commands

**Comment by PartialDevAdmin** [comment]
First error!

```
Fatal error: Uncaught ReflectionException: Class Reflar\ThemeChanger\Listeners\AddThemesApi does not exist in C:\xampp\htdocs\vendor\illuminate\container\Container.php:741 Stack trace: #0 C:\xampp\htdocs\vendor\illuminate\container\Container.php(741): ReflectionClass->__construct('Reflar\\ThemeCha...') #1 C:\xampp\htdocs\vendor\illuminate\container\Container.php(631): Illuminate\Container\Container->build('Reflar\\ThemeCha...', Array) #2 C:\xampp\htdocs\vendor\flarum\core\src\Foundation\Application.php(514): Illuminate\Container\Container->make('Reflar\\ThemeCha...', Array) #3 C:\xampp\htdocs\vendor\illuminate\events\Dispatcher.php(149): Flarum\Foundation\Application->make('Reflar\\ThemeCha...') #4 C:\xampp\htdocs\vendor\illuminate\events\Dispatcher.php(135): Illuminate\Events\Dispatcher->resolveSubscriber('Reflar\\ThemeCha...') #5 C:\xampp\htdocs\workbench\theme-changer\bootstrap.php(20): Illuminate\Events\Dispatcher->subscribe('Reflar\\ThemeCha...') #6 [internal function]: Flarum\Extension\ExtensionServiceProvider->Refl in C:\xampp\htdocs\vendor\illuminate\container\Container.php on line 741
```

For some reason, it appears like it doesn't think the AddThemesApi file exists :thinking:

**Comment by PartialDevAdmin** [comment]
NVM, fixed that issue

**Comment by PartialDevAdmin** [comment]
I have a migrations file created now

**Comment by PartialDevAdmin** [comment]
@CDK2020 Ok, base code is in. There are no errors, but it does not work. Can you see any errors in the API or the JS?

**Comment by PartialDevAdmin** [comment]
Migrations do work, just so you know

**Comment by PartialDevAdmin** [comment]
What did you mean by `you don't have a theme modal class`

**Comment by CDK2020** [comment]
You hadn't pushed something yet, no problems.

Also the repository should be in a repository folder and same with the validator 

You don't have to tag me if you don't want, I get the notifications regardless

I have no idea what wouldn't be working you'll have to try to debug yourself

**Comment by PartialDevAdmin** [comment]
```
GET http://localhost/api/Themes?page%5Boffset%5D=0&page%5Blimit%5D=20&sort=-time

{
  "errors": [
    {
      "status": "404",
      "code": "route_not_found"
    }
  ]
}
```

is the debug code. What would the API route be?

**Comment by CDK2020** [comment]
Why do you have filters in the URL? You can't just copy and paste code an expect it to work. Try to debug it yourself

**Comment by PartialDevAdmin** [comment]
Ok

**Comment by PartialDevAdmin** [comment]
What is the default API route?

**Comment by luceos** [comment]
tip: URL's need to be lowercase, `api/Themes`

> default API route

There is no default api route. You can register API routes using the `ConfigureApiRoutes` event (if I'm not mistaken).
